examples/sound_effects/SoundEffects.cpp

```diff
@@ -758,12 +758,19 @@ class BiquadFilter : public Processing
         music.setEffectProcessor(
             [coefficients,
              enabled = getEnabled(),
-             state   = std::vector<State>(music.getChannelCount())](const float*  inputFrames,
-                                                                  unsigned int& inputFrameCount,
-                                                                  float*        outputFrames,
-                                                                  unsigned int& outputFrameCount,
-                                                                  unsigned int  frameChannelCount) mutable
+             state   = std::vector<State>()](const float*  inputFrames,
+                                           unsigned int& inputFrameCount,
+                                           float*        outputFrames,
+                                           unsigned int& outputFrameCount,
+                                           unsigned int  frameChannelCount) mutable
             {
+                // IMPORTANT: The channel count of the audio engine currently sourcing data from this sound
+                // will always be provided in frameChannelCount, this can be different from the channel count
+                // of the audio source so make sure to size your buffers according to the engine and not the source
+                // Ensure we have as many state objects as the audio engine has channels
```

Comment:
Maybe it would make sense to add this to the already length documentation of the function somehow.

Reply:
Added to documentation.